src/main/resources/db/migration/V1_68__create_utilisation_report.sql

```diff
@@ -0,0 +1,307 @@
+    and p.loc_code = p_loc_code
+    and ((jp.status = 10 and jp.transfer_date >= p_start_date)
+           or (jp.status <> 10 and (j.completion_date is null or j.completion_date >= p_start_date)))
+    group by jp.juror_number
+    having (least(coalesce(min(a.attendance_date), min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end)),
+               min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end)) <= p_end_date);
+END;
+$function$
+;
+
+
+-- function to return list of appearance for jurors within report window
+CREATE OR REPLACE FUNCTION juror_mod.util_report_appearance_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+  	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	a.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+	null::date,
+	null::date,
+	null::character varying,
+	null::date,
+	null::date,
+	a.attendance_date,
+	true,
+	a.sat_on_jury,
+	true,
+	null::boolean
+	from juror_mod.appearance a
+	where a.attendance_date between p_start_date and p_end_date
+	and (a.non_attendance is null or a.non_attendance = false)
+	and (a.no_show is null or a.no_show = false)
+	and a.loc_code= p_loc_code
+	order by a.juror_number, a.attendance_date;
+
+END;
+$function$
+;
+
+
+-- function to return juror participation in a trial within the report window
+CREATE OR REPLACE FUNCTION juror_mod.util_trial_participation_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+   	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	jt.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+		null::date,
+	null::date,
+	jt.trial_number,
+	jt.empanelled_date as jury_from,
+	coalesce (jt.return_date, p_end_date) as jury_to,
+	report_dates.report_date,
+	true,
+	case when report_dates.report_date between jt.empanelled_date and coalesce (jt.return_date, p_end_date) then true else false end as active_date,
+	null::boolean,
+	null::boolean
+	from juror_mod.juror_trial jt
+	left join (select * from juror_mod.util_report_report_days_list(p_loc_code, p_start_date, p_end_date)) as report_dates
+	on report_dates.report_date between jt.empanelled_date and coalesce (jt.return_date, p_end_date) -- if no return date then assume still on trial for now
```

Comment:
this confused me 😅  but after a little test it looks to produce the desired result 👌

Reply:
> this confused me 😅 but after a little test it looks to produce the desired result 👌

I too am confused. The use of a LEFT join to me means everything from juror_trial (left) plus the rows from the report date list (right) that match the dates in juror_trial. I must have that wrong. Can we discuss please.

Reply:
This can be an inner join actually... we should have only dates that are in the report window - results are the same